EIPS/eip-3220.md

```diff
@@ -0,0 +1,121 @@
+
+### Definition of crosschain id registration smart contract
+
+The crosschain registration smart contract provides a registration service to crosschain id and its associated meta data.  The smart contract should support the following meta data.
+
+|Metadata Name| Description  | Examples|
+|-------------|--------------|---------|
+|Crosschain Id|32 byte hex id|0xabc. See the crosschain id section for details|
+|Long Name|A long naname describing the blockchain.  Maximum length is 80 characters.|This long name could be **Ethereum Mainnet**, **Bitcoin Cash**, **Wanchain Mainnet** etc|
+|Short Name|An abbreviated name for the blockchain. Maximum length is four characters.|For public blockchains, this will match the standard blockchain exchange name. For example, **ETC**, **BTC**.|
+|Nickname|Nickname of the chain. Maximum length is 10 characters.|Names such as **Rinkeby** or **Ropsten**.|
+|Chain Type|A meta data for the type of the blockchains such as public or private, permissioned or permissionless| Private:permissioned|
+|Chain Category|A parameter to identify whether this is mainnet, or testnet|"mainnet", "testnet"|
+|Chain IP addresses and Service Provider Id|A set of IP addresses of static nodes that can be contacted to submit a crosschain transaction, and the associated name information for the company providing the service.|125.125.125.125, 10.10.20.20|
+|Name Service Type|The contract naming service available on the blockchain.|The type of service, for instance **ENS**.|
+|Name Service Address|The address of the name service contract on the blockchain.| |
+
+### Crosshchain Identifier Registration Contract
+
+To register crosschian/blockchain id, there should be grace period for comments before it become official and verified.  The registration service should support the following functions
+
+* RegisterBlockchainId
+* ModifyName
+* ModifyParameters
+* VerifyBlockchainId
+* QueryBlockChainId
+* LookupBlockchainId
+* QueryBlockchainOwner
+* QueryBlockchainName(Blockchain-id, NameType)
+* QueryParentBlockchainId
+
+## Rationale
+
+With the success of Bitcoin and Ethereum, various blockchains such as EOS, Ripple, Litecoin, Besu, Wanchain and the like have been developed and are growing at a fast pace.  There are also other private and consortium blockchains such as Hyperledger Fabric, Hyperledger Besu, Stellar, Corda, Quorum that only allow nodes with permitted identities to join the blockchain network.  The growth of public and private blockchains imposes
+challenges for inter-chain interoperability, particularly when these chains are
+heterogeneous and incompatible.
```

Comment:
Very weird newlines here.  Recommend a newline after every sentence.  Markdown will turn that into one paragraph, and it makes it much easier to give targeted feedback on specific lines.

Reply:
Modified